python/src/nanoarrow/_lib.pyx

```diff
@@ -894,8 +894,11 @@ cdef class ArrayStream:
             return array
 
     def __iter__(self):
-        while True:
-            yield self.get_next()
+        try:
```

Comment:
I _think_ this is also the use case for `yield from` in Python

Reply:
I was also refreshing my Python knowledge on this, and so it's actually in the `__next__` method that a StopIteration needs to be raised, not in `__iter__`

So another way to implement this would be:

```
def __iter__(self):
    return self

def __next__(self):
    return self.get_next()
```

(but the current one works fine as well)